### ISSUE TITLE
RFC: Add a plugin for generating an express API gateway from an environment

### ISSUE THREAD

**Issue by mattpocock** [mattpocock/sextant#9]
An awesome use for Sextant could be generating code for a Lambda endpoint. Let's imagine you have a consistent environment name of 'lambda' across multiple services (glossary here #7). This lambda can be accessed from another consistently named environment, the 'frontend'.

So each service could map to a `/path` (Get Users becomes `/get-users`) which each takes a POST, and the event as an application/json. It then must send back an event described in Sextant. This might be implemented like this:

```js
import express from 'express';
import { buildExpressApi } from './sextant-generated-code';

const app = express();

const lambdaEndpoints = buildExpressApi({
  // req.body would be typed to the input event, and req.json would be typed to the output
  getUsers: (req, res) => {},
  someOtherService: (req, res) => {},
});

lambdaEndpoints.forEach(({ handler, path }) => {
  app.post(path, handler)
});
```

**Comment by mattpocock** [comment]
A final thought. Not only could this generate the API route, but it could also generate an SDK which _talks_ to the API route.

**Comment by ditorojuan** [comment]
Love this Idea. But here we are kinda blocked by the plug-in system, right?

**Comment by mattpocock** [comment]
For sure. See #10

**Comment by mattpocock** [comment]
I like the idea of creating issues for plugin ideas, though. An RFC process for what people want from plugins will be crucial to getting them right.

**Comment by mattpocock** [comment]
Fixed in #15
